address.c

```diff
@@ -56,16 +58,24 @@ bool address_check_prefix(const uint8_t *addr, uint32_t address_type)
 #if USE_ETHEREUM
 #include "sha3.h"
 
-void ethereum_address_checksum(const uint8_t *addr, char *address)
+void ethereum_address_checksum(const uint8_t *addr, char *address, int rskip60, int chain_id)
 {
 	const char *hex = "0123456789abcdef";
 	for (int i = 0; i < 20; i++) {
 		address[i * 2]     = hex[(addr[i] >> 4) & 0xF];
 		address[i * 2 + 1] = hex[addr[i] & 0xF];
 	}
 	address[40] = 0;
+
+	char prefixedAddress[50];
+
+	if(rskip60)
+		snprintf(prefixedAddress, sizeof(prefixedAddress), "%d0x%s", chain_id, address);
+	else
+		snprintf(prefixedAddress, sizeof(prefixedAddress), "%s", address);
+
 	uint8_t hash[32];
-	keccak_256((const uint8_t *)address, 40, hash);
+	keccak_256((const uint8_t *)prefixedAddress, strlen(prefixedAddress), hash);
```

Comment:
Rather than copying the address into a new buffer, use `keccak_Update`. For example,

```c
SHA3_CTX ctx;
keccak_256_Init(&ctx);
keccak_Update(&ctx, prefix, strlen(prefix));
keccak_Update(&ctx, (const uint8_t *)address, 40);
keccak_Final(&ctx, hash);
```